registry/api/errcode/register.go

```diff
@@ -33,14 +33,25 @@ var (
 		HTTPStatusCode: http.StatusMethodNotAllowed,
 	})
 
+	// ErrorCodeUnauthenticated is returned if a request requires
+	// authentication.
+	ErrorCodeUnauthenticated = Register("errcode", ErrorDescriptor{
+		Value:   "UNAUTHENTICATED",
+		Message: "authentication required",
+		Description: `The access controller was unable to authenticate
+		the client. Often this will be accompanied by a
+		Www-Authenticate HTTP response header indicating how to
+		authenticate.`,
+		HTTPStatusCode: http.StatusUnauthorized,
+	})
+
 	// ErrorCodeUnauthorized is returned if a request is not authorized.
 	ErrorCodeUnauthorized = Register("errcode", ErrorDescriptor{
 		Value:   "UNAUTHORIZED",
```

Comment:
Isn't this a little confusing to map forbidden to unauthorized and unauthenticated to unauthorized? It might be better if we follow this:

|Error Code|HTTP status| Action | Description | WWW-Authenticate |
|-----------------|----------|------------------|------------|---|
| UNAUTHORIZED | 401 Unauthorized | Authorize | Missing authentication. Client must authorize. | Yes |
| DENIED | 403 | Terminal | No permission to access resource. Client should not proceed | No |

This way we don't have to change the mapping between status and error code.